apps/aecore/src/aec_chain_state.erl

```diff
@@ -920,15 +918,15 @@ calculate_gas_fee(Calls) ->
 
 apply_micro_block_transactions(Node, FeesIn, Trees) ->
     Txs = db_get_txs(hash(Node)),
-    Height = node_height(Node),
-    Version = node_version(Node),
     TotalFees = lists:foldl(
                   fun(SignedTx, AccFee) ->
                           Fee = aetx:fee(aetx_sign:tx(SignedTx)),
                           AccFee + Fee
                   end, FeesIn, Txs),
-
-    case aec_block_micro_candidate:apply_block_txs_strict(Txs, Trees, Height, Version) of
+    KeyHeader = db_get_header(prev_key_hash(Node)),
+    Env = aetx_env:tx_env_from_key_header(KeyHeader, prev_key_hash(Node),
+                                          node_time(Node), prev_hash(Node)),
```

Comment:
I see this is in scope of ["Set timestamp from current block timestamp in contract state."](https://www.pivotaltracker.com/n/projects/2159793/stories/159522630).

I believe a note in spec (protocol repo - [here](https://github.com/aeternity/protocol/blob/108d94fc543c79b78a97608d512bba3963f6c6c5/contracts/aevm.md#chain-specific-instructions)?) shall be added mentioning that `TIMESTAMP` returns the timestamp of the current microblock - not the *keyblock* (it may have been keyblock) and not the *previous* microblock/keyblock.

Reply:
Described in https://github.com/aeternity/protocol/pull/209